FTBFS lftp

### Intended Version

4.8.4

### Target Branch

stable

### Description

Build log:

```
Making install in src
make[2]: Entering directory '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abbuild/src'
make[3]: Entering directory '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abbuild/src'
 /usr/bin/mkdir -p '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abdist/usr/lib'
 /usr/bin/mkdir -p '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abdist/usr/bin'
 /usr/bin/mkdir -p '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abdist/usr/share/lftp'
 /bin/sh ../libtool --silent   --mode=install /usr/bin/install -c   liblftp-tasks.la liblftp-jobs.la '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abdist/usr/lib'
 /usr/bin/install -c /var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/src/lftpget '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abdist/usr/bin'
 /usr/bin/install -c /var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/src/import-ncftp /var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/src/import-netscape /var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/src/verify-file /var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/src/convert-mozilla-cookies /var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/src/xdg-move '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abdist/usr/share/lftp'
libtool: warning: remember to run 'libtool --finish /usr/lib'
 /usr/bin/mkdir -p '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abdist/usr/bin'
make  install-data-hook
  /bin/sh ../libtool --silent   --mode=install /usr/bin/install -c lftp '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abdist/usr/bin'
make[4]: Entering directory '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abbuild/src'
rmdir /var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abdist/usr/lib/lftp/4.8.4 2>/dev/null || :
make[4]: Leaving directory '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abbuild/src'
make[3]: Leaving directory '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abbuild/src'
make[2]: Leaving directory '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abbuild/src'
Making install in po
make[2]: Entering directory '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abbuild/po'
/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abdist/usr/share
make[2]: /var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abdist/usr/share: Permission denied
make[2]: *** [Makefile:304: install-data-yes] Error 127
make[2]: Leaving directory '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abbuild/po'
make[1]: *** [Makefile:1695: install-recursive] Error 1
make[1]: Leaving directory '/var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4/abbuild'
make: *** [Makefile:2000: install] Error 2
[ERROR]: Build error detected ^o^
In file included from /usr/bin/autobuild:65
                 from /usr/lib/autobuild3/lib/base.sh:118
                 from /usr/lib/autobuild3/lib/base.sh:122
                 from /usr/lib/autobuild3/proc/50-build_exec.sh:43
/usr/lib/autobuild3/build/10-autotools.sh:92: In function (unknown):
/usr/lib/autobuild3/build/10-autotools.sh:92: error: command exited with 2
>  92 |                 || abdie "Failed to install binaries: $?."

autobuild encountered an error and couldn't continue.
Failed to install binaries: 2.
------------------------------autobuild 3------------------------------
Go to ‘https://github.com/AOSC-Dev/autobuild3’ for more information on this error.
[ERROR]: Error when building lftp.
Build folder: /var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/acbs/main.py", line 253, in build_sequential
    invoke_autobuild(task, build_dir)
  File "/usr/lib/python3.10/site-packages/acbs/utils.py", line 199, in invoke_autobuild
    start_build_capture(env_dict, build_dir)
  File "/usr/lib/python3.10/site-packages/acbs/utils.py", line 150, in start_build_capture
    raise RuntimeError('autobuild3 did not exit successfully.')
RuntimeError: autobuild3 did not exit successfully.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/acbs/main.py", line 124, in build
    self.build_sequential(build_timings, packages)
  File "/usr/lib/python3.10/site-packages/acbs/main.py", line 259, in build_sequential
    raise RuntimeError(
RuntimeError: Error when building lftp.
Build folder: /var/cache/acbs/build/acbs.4q0bjdj2/lftp-4.8.4
[INFO]: ACBS is trying to save your build status...
[INFO]: ... saved to /tmp/1692290856.acbs-ckpt

[CRIT]: Oops! RuntimeError: Build error.
Use `acbs-build --resume /tmp/1692290856.acbs-ckpt` to resume after you sorted out the situation.
error: Build failed with status: 1
info: Ciel created a check-point: ./STATES/lftp-1692290856.ciel-ckpt

```

--- a/app-network/lftp/autobuild/patches/0001-Fedora-lftp-4.9.2-cdefs.patch
+++ b/app-network/lftp/autobuild/patches/0001-Fedora-lftp-4.9.2-cdefs.patch
@@ -1,0 +1,13 @@
+diff --git a/lib/libc-config.h.old b/lib/libc-config.h
+index 1300c3a..7537bab 100644
+--- a/lib/libc-config.h.old
++++ b/lib/libc-config.h
+@@ -156,7 +156,7 @@
+ #undef __warndecl
+ 
+ /* Include our copy of glibc <sys/cdefs.h>.  */
+-#include <cdefs.h>
++#include <sys/cdefs.h>
+ 
+ /* <cdefs.h> __inline is too pessimistic for non-GCC.  */
+ #undef __inline

--- a/app-network/lftp/autobuild/prepare
+++ b/app-network/lftp/autobuild/prepare
@@ -1,0 +1,2 @@
+abinfo "Fixing mkdir detection ..."
+export MKDIR_P="/usr/bin/mkdir -p"

--- a/app-network/lftp/spec
+++ b/app-network/lftp/spec
@@ -1,5 +1,4 @@
-VER=4.8.4
+VER=4.9.2
 SRCS="tbl::https://lftp.yar.ru/ftp/lftp-${VER}.tar.xz"
-CHKSUMS="sha256::4ebc271e9e5cea84a683375a0f7e91086e5dac90c5d51bb3f169f75386107a62"
-REL=1
+CHKSUMS="sha256::c517c4f4f9c39bd415d7313088a2b1e313b2d386867fe40b7692b83a20f0670d"
 CHKUPDATE="anitya::id=1555"

